charts/seed-monitoring/charts/core/charts/prometheus/templates/config.yaml

```
@@ -547,6 +547,26 @@ data:
         action: keep
         regex: ^{{ .Release.Namespace }}$
 
+{{- if  (index .Values.rules.optional "hvpa" ).enabled }}
```

I found the issue. There needs to be a newline before and after the block.
```suggestion
{{ if  (index .Values.rules.optional "hvpa" ).enabled -}}
    - job_name: 'hvpa-controller'
      kubernetes_sd_configs:
      - role: endpoints
        namespaces:
          names: [ garden ]
      relabel_configs:
      - source_labels:
        - __meta_kubernetes_service_name
        - __meta_kubernetes_endpoint_port_name
        - __meta_kubernetes_namespace
        action: keep
        regex: hvpa-controller;metrics;garden
      metric_relabel_configs:
{{ include "prometheus.keep-metrics.metric-relabel-config" .Values.allowedMetrics.hvpa | indent 6 }}
      - source_labels: [ namespace ]
        action: keep
        regex: ^{{ .Release.Namespace }}$
{{ end }}
```

---

Thanks @wyb1. I have made the suggested changes.

---

Looks like I made a mistake and this doesn't solve the problem. `{{ if  (index .Values.rules.optional "hvpa" ).enabled -}}` will remove the indentation of the job name and cause it to be invalid yaml. `{{ if  (index .Values.rules.optional "hvpa" ).enabled }}` also doesn't work.

---

I am not sure why the original yaml isn't working. The `elasticsearch` block below this also has the similar template.

---

I have changed the hvpa-controller port from 8080 to 5769. The change still needs to be made in hvpa-controller repo. I have already raised the PR. Once that is merged, i think the original template would work. Meanwhile I have made changes to this PR so that it becomes independent of hvpa-controller's default port.

---

@ggaurav10 what's the status now that the PR is merged?

---

I have added the command line option in the gardener's hvpa chart itself. So this change is independent of the PR on hvpa-controller. This PR is ready for final review/merge.

---

OK, thanks. @wyb1 can you please check again?

---

I still cannot reconcile without `error converting YAML to JSON`.